src/Motions/PairMatch.ts

```diff
@@ -0,0 +1,90 @@
+                                       "<"  : ">",
+                                       ">"  : ">",
+                                       "("  : ")",
+                                       ")"  : ")",
+                                       "["  : "]",
+                                       "]"  : "]",
+                                       "{"  : "}", 
+                                       "}"  : "}",
+                                      };
+                                 
+    private static mapToOpeningChar = {"'"  : "'",
+                                       "\"" : "\"", 
+                                       ">"  : "<",
+                                       "<"  : "<",
+                                       ")"  : "(",
+                                       "("  : "(",
+                                       "]"  : "[",
+                                       "["  : "[",
+                                       "}"  : "{",
+                                       "{"  : "{"
+                                      };
+   
+    /** Go to the first opening char (exclusive) corresponding to 'args.character'  */
+    static matchOpening(args: {character: string}): MotionPairMatch {        
+        const obj = new MotionPairMatch();
+        obj.character = args.character;
+        obj.direction = MotionPairMatchDirection.OPENING;
+        return obj;
+    }
+    
+    /** Go to the first closing char (exclusive) corresponding to 'args.character'  */
+    static matchEnding(args: {character: string}): MotionPairMatch {        
+        const obj = new MotionPairMatch();
+        obj.character = args.character;
+        obj.direction = MotionPairMatchDirection.CLOSING;    
+        return obj;
+    }
+
+    apply(from: Position, option: {isInclusive?: boolean} = {}): Position {
+        from = super.apply(from);
+        const activeTextEditor = window.activeTextEditor;
+
+        if (! activeTextEditor || this.direction === undefined || ! this.character) {
+            return from;
+        }
+
+        const document = activeTextEditor.document;
+        let toLine = from.line;
+        let toCharacter = from.character;
+        let targetText:string = document.lineAt(toLine).text;
+
+        if (this.direction === MotionPairMatchDirection.OPENING) {
+             targetText = targetText
+                .substr(0, toCharacter)
+                .split('').reverse().join('');
+
+            var openingChar = MotionPairMatch.mapToOpeningChar[this.character];
+            const offset = targetText.indexOf(openingChar);
+
+            if (!!~offset) {
+                toCharacter -= offset;
+            }
+        }
+
+        else if (this.direction === MotionPairMatchDirection.CLOSING) {
+            targetText = targetText.substr(toCharacter + 1);
+            var closingChar = MotionPairMatch.mapToClosingChar[this.character];
+            const offset = targetText.indexOf(closingChar);
```

Comment:
This won't work, because since we're not doing any sort of balancing, we can't be sure that the next occurrence of `closingChar` actually corresponds to the the opening character we're starting looking from.  For example:

    console.log(foo(bar("baz")));

If the cursor is positioned over the first "o" in "foo", then pressing `ci)` leaves us with unbalanced parentheses:

    console.log()));

Same goes for `MotionPairMatchDirection.OPENING`

vscode has "editor.action.jumpToBracket", but we may not want to use it since it creates an entry in the editor navigation history.  Would be nice if vscode would expose some language and text services so we could piggyback off the existing implementation for stuff like this.  What we really want is to be able to ask "What's the position that would be computed for the 'jumpToBracket' command?"

In the meantime, though, it would probably be beneficial *not* to wait for this kind of API to show up and instead write a quick and dirty implementation to do the whole thing ourselves while we wait.

It would be re-used for the `%` motion.